backend/src/middleware/detection.js

```
@@ -0,0 +1,541 @@
+};
+
+/**
+ * Detect schema from API Bearer token for API-Auth-Server mode
+ * Sets API_CLIENT pool context (for server-to-server calls)
+ */
+export const detectSchemaFromApiToken = async (req, res, next) => {
+   try {
+      const authHeader = req.headers.authorization;
+
+      if (authHeader && authHeader.startsWith("Bearer ")) {
+         const token = authHeader.substring(7);
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)